libnfs4acl/nfs4_set_acl.c

```diff
@@ -33,37 +33,50 @@
 #include <sys/stat.h>
 #include <sys/errno.h>
 #include <attr/xattr.h>
+#include <err.h>
 #include <unistd.h>
 #include <stdio.h>
 #include "libacl_nfs4.h"
 
-int nfs4_set_acl(struct nfs4_acl *acl, const char *path)
+int nfs4_acl_set_file(struct nfs4_acl *acl, const char *path)
 {
 	int res = 0;
 	char *xdrbuf = NULL;
 
 	res = acl_nfs4_xattr_pack(acl, &xdrbuf);
 	if (res <= 0) {
-		fprintf(stderr, "Failed to populate xattr from nfs4acl\n");
-		goto out_free;
+		warnx("nfs4_acl_set_file() failed");
+		free(xdrbuf);
+		return (res);
 	}
 
-	res = setxattr(path, ACL_NFS4_XATTR, (char *)xdrbuf, res, XATTR_REPLACE);
-	if (res < -10000) {
-		fprintf(stderr,"An internal NFS server error code (%d) was returned; this should never happen.\n",res);
-		goto out_free;
-	} else if (res < 0) {
-		if (errno == EOPNOTSUPP)
-			fprintf(stderr,"Operation to set ACL not supported.\n");
-		else if (errno == ENOATTR)
-			fprintf(stderr,"ACL Attribute not found on file.\n");
-		else if (errno == EREMOTEIO)
-			fprintf(stderr,"An NFS server error occurred.\n");
-		else
-			perror("Failed setxattr operation");
+	res = setxattr(path, ACL_NFS4_XATTR, xdrbuf, res, XATTR_REPLACE);
+	if (res < 0) {
+		warnx("nfs4_acl_set_file() failed");
+	}
+
+	free(xdrbuf);
+	return (res);
+}
+
+
+int nfs4_acl_set_fd(struct nfs4_acl *acl, int fd)
+{
+	int res = 0;
+	char *xdrbuf = NULL;
+
+	res = acl_nfs4_xattr_pack(acl, &xdrbuf);
+	if (res <= 0) {
```

Comment:
I haven't checked `acl_nfs4_xattr_pack` but curious if `res` should be `<= 0` ??

Reply:
This check was from original nfs4-acl-tools. A zero-sized xattr is also an error condition.